mypy/typeshed/stdlib/json/__init__.pyi

```diff
@@ -1,14 +1,17 @@
 from _typeshed import SupportsRead, SupportsWrite
 from collections.abc import Callable
-from typing import Any
+from typing import Any, overload
+from typing_extensions import TypeAlias
 
 from .decoder import JSONDecodeError as JSONDecodeError, JSONDecoder as JSONDecoder
 from .encoder import JSONEncoder as JSONEncoder
 
 __all__ = ["dump", "dumps", "load", "loads", "JSONDecoder", "JSONDecodeError", "JSONEncoder"]
 
+_JsonType: TypeAlias = dict[str, _JsonType] | list[_JsonType] | float | int | bool | str | None
```

Comment:
I intentionally included bool, im not completely final on my choice, but I think it's less confusing, and the subtype of int isn't the most straightforward thing in the world

Reply:
> also, you could use [`optype.json.Value`](https://github.com/jorenham/optype/blob/master/optype/json.py#L28) instead, which is literally equivalent to this, but I'll take this opportunity to shamelessly self-promote my own package either way 😎 

can we use it in typeshed? or will it cause circular import problems?

happy to add it, just need to think about any complications it might introduce

Reply:
I thought that only worked for stub-only packages 🤔

Reply:
But if not, it would be really cool to be able to have all those `optype.Can*` protocols available when writing stubs

Reply:
> I thought that only worked for stub-only packages

i don't follow, what are you refering to?

> would be really cool to be able to have all those optype.Can* protocols available when writing stubs

do you mean include optype in typeshed itself? or making it a mandatory dependency of basedmypy?

Reply:
> can we use it in typeshed?

I thought you were talking about including it in the same way as `typing_extensions` and `_typeshed`

Reply:
> I intentionally included bool, im not completely final on my choice, but I think it's less confusing, and the subtype of int isn't the most straightforward thing in the world

I used to to that as well, but the issue with such things is that in the case that type-checkers report errors that are related to it, the error messages become more difficult to read.
But that being said, in an ideal word that shouldn't be something that you should have to worry about here at all, as it's not the responsibility of the typer 🤷🏻

Reply:
Although I apparently am a hypocrite 🦛
https://github.com/jorenham/optype/blob/6a153d9c4dde6fefa6bce4a6c679f3e5135cb50b/optype/typing.py#L54

Reply:
I'll raise an issue about putting optype in the std typeshed